cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -3077,26 +3027,9 @@ String getInternalServiceHostname(String serviceName)    {
 
         private final Future<ReconciliationState> getKafkaExporterDescription() {
             Promise<ReconciliationState> promise = Promise.promise();
-
-            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").<ReconciliationState>executeBlocking(
-                future -> {
-                    try {
-                        this.kafkaExporter = KafkaExporter.fromCrd(kafkaAssembly, versions);
-                        this.exporterDeployment = kafkaExporter.generateDeployment(pfa.isOpenshift(), imagePullPolicy, imagePullSecrets);
-
-                        future.complete(this);
-                    } catch (Throwable e) {
-                        future.fail(e);
-                    }
-                }, true,
-                res -> {
-                    if (res.succeeded()) {
-                        promise.complete(res.result());
-                    } else {
-                        promise.fail(res.cause());
-                    }
-                }
-            );
+            this.kafkaExporter = KafkaExporter.fromCrd(kafkaAssembly, versions);
+            this.exporterDeployment = kafkaExporter.generateDeployment(pfa.isOpenshift(), imagePullPolicy, imagePullSecrets);
+            promise.complete(this);
```

Comment:
Since we don't do anything asynchronous here, you can probably just do `return Future.succeededFuture(this);` and not use the `Promise` at all.